### DISCUSSION TITLE
Rauc behavior on power loss and network loss

### DISCUSSION THREAD

**Discussion by lmussier** [rauc/rauc#1385] [Q&A] [answered]
Hi,
I'm struggling to understand the expected behavior of RAUC in some scenarios we've encountered.
Given the discussions on the following GitHub links: [PR 755](https://github.com/rauc/rauc/pull/755), [Issue 768](https://github.com/rauc/rauc/issues/768), and [Discussion 1023](https://github.com/rauc/rauc/discussions/1023).
If it helps, we are using the HTTP streaming installation feature.

1. Network loss:
Based on my understanding, RAUC will attempt to retry downloading from the server several times (typically 3 times) in case of network loss. After these retries, if the download is unsuccessful, the installation will be marked as failed.

1. Power outage / reboot:
I couldn't find specific information on how RAUC handles power outages or reboots. It's unclear if RAUC remembers the installation state or if it's stateless, requiring a manual retrigger of the installation after a reboot or power loss.

Lastly, is there a safe way to check if RAUC is actively processing something? We were considering using the status command until we came across this issue: [Issue 416](https://github.com/rauc/rauc/issues/416#issuecomment-471625190).

**Comment by jluebbe** [comment, marked as answer]
> Hi, I'm struggling to understand the expected behavior of RAUC in some scenarios we've encountered. Given the discussions on the following GitHub links: [PR 755](https://github.com/rauc/rauc/pull/755), [Issue 768](https://github.com/rauc/rauc/issues/768), and [Discussion 1023](https://github.com/rauc/rauc/discussions/1023). If it helps, we are using the HTTP streaming installation feature.
> 
> 1. Network loss:
>    Based on my understanding, RAUC will attempt to retry downloading from the server several times (typically 3 times) in case of network loss. After these retries, if the download is unsuccessful, the installation will be marked as failed.

That's correct. Adaptive updates could be enhanced so that data which was written to the target slot before the network loss doesn't need to be downloaded again. That change would be relatively simple.

> 2. Power outage / reboot:
>    I couldn't find specific information on how RAUC handles power outages or reboots. It's unclear if RAUC remembers the installation state or if it's stateless, requiring a manual retrigger of the installation after a reboot or power loss.

A power loss or crash implicitly aborts the installation. As the active slots are never touched during installation, this means that a system reset during an installation is no different from a normal reset. You then can trigger the installation again.

> Lastly, is there a safe way to check if RAUC is actively processing something? We were considering using the status command until we came across this issue: [Issue 416](https://github.com/rauc/rauc/issues/416#issuecomment-471625190).

You should use the D-Bus property `Operation`: https://github.com/rauc/rauc/blob/master/src/de.pengutronix.rauc.Installer.xml#L84, which is currently either `idle` or `installing`.

**Reply by lmussier** [reply]
Perfect thanks !
